modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/dateRangeFilter.js

```diff
@@ -215,5 +205,8 @@ DateRangeFilter.propTypes = {
 	}),
 };
 
-export {getSelectedItemsLabel, getOdataString};
-export default DateRangeFilter;
+export default {
+	Component: DateRangeFilter,
+	getSelectedItemsLabel,
+	getOdataString
+};
```

Comment:
This pattern for an export looks weird to me, but it does make sense. I'm used to exports that are components or utilities. This is an object, a "class" of sorts, containing "methods". Maybe we want to refactor internally and export a single `class` here?

Reply:
A single class is weird too because `getSelectedItemsLabel` and `getOdataString` would need to be static methods of a class that is implementing a UI component which is a smell too, because the methods are not dependent on any UI stuff (they are quite abstract).

We could also do it like this:

```javascript
export DateRangeFilter as Component;
export getSelectedItemsLabel;
export getOdataString;
```

and then import in `Filter.js` like this:

```javascript
import * as dateFilter from './dateFilter';
```

However, there's an implicit contract between `Filter.js` and the filter implementation which consists in each filter implementation exporting the two methods and the component class. Thus, we may want to create an interface in TypeScript to formally describe and check it, like this:

```javascript
interface FilterImplementation {
   Component: ....,
   getSelectedItemsLabel: (...) => ...,
   ...
}
```

And then apply it to the exported object of the filter implementation.

This way, if the contract changes in the future, the typescript compiler will warn you if you haven't adapted things properly in any of the filter implementations.

In any case, I believe this would look better than what I wrote while still let us type the contract:

```javascript
export {
	Component: DateRangeFilter,
	getSelectedItemsLabel,
	getOdataString
};
```

Reply:
```js
interface FilterImplementation {
   Component: ....,
   getSelectedItemsLabel: (...) => ...,
   ...
}
```
Looks great! :+1: 

With this, it would probably make sense to name the file `DateRangeFilterImplementation.js`

Reply:
`FilterImplementation` is a good idea, indeed, it would be a public API and so live in `@liferay/js-api/data-set` for usage by the client extension code.

Alternatively, we can create a more flexible api for our internal filters, yet export the bare minimum parts to the public API

Reply:
Yes, we need two types (even if we decide that the internal extends the public one). This is because `ClientExtensionFilterImplementation` would implement the internal `FilterImplementation` while the CX itself would implement the public `FDSFilterCX` (whatever we decide to call it) interface.

Reply:
`FilterImplementation` would live in DXP source code
`FDSFilterCX` (whatever we decide to call it) would live in `@liferay/js-api`